velox/functions/sparksql/tests/SparkCastExprTest.cpp

```diff
@@ -107,7 +107,6 @@ TEST_F(SparkCastExprTest, date) {
        "1970-01-2",
        "1970-1-02",
        "+1970-01-02",
-       "-1-1-1",
```

Comment:
not sure I followed the exact Spark semantic, but could we add more test with one, two, three, four and more year digits to ensure it does what it is expected?

Reply:
added.